### PR TITLE
Fix deadmin on join cvar for nukies

### LINKED ISSUE #19271: Deadmin on join cvar doesn't deadmin you if you roll nukie
Didn't test it myself, it was just reported to me

### PR DESCRIPTION
**Pull request by LankLTE**
## About the PR

Fixes #19271 . Deadmins the player if the cvar is active and they're an admin.

## Technical details

This was broken as RulePlayerSpawningEvent is broken immediately as the spawning event is called, within SpawnPlayers whereas the cvar was used in SpawnPlayer after the event is called. I don't just iterate through it in SpawnPlayers (doing so in  in the nukeopsrulesystem instead) as it still needs to exist in SpawnPlayer as I believe that method is used elsewhere, but SpawnPlayers eventually calls SpawnPlayer. I'm sure it wouldn't break anything but it made more sense to do it here. I believe nukeops are the only antag that currently don't use the standard station spawning system. 

I'm not certain how the webhook works so maybe someone smarter than myself can inform me.

### CODE REVIEW

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 1]
```diff
@@ -923,6 +928,10 @@ private void SpawnOperatives(int spawnCount, List<IPlayerSession> sessions, bool
                 _mindSystem.SetUserId(newMind, session.UserId);
                 _roles.MindAddRole(newMind, new NukeopsRoleComponent { PrototypeId = spawnDetails.Role });
 
+                // Automatically de-admin players who are being made nukeops
+                if (_cfg.GetCVar(CCVars.AdminDeadminOnJoin) && _adminManager.IsAdmin(session))
+                    _adminManager.DeAdmin(session);
+
```
Comment (metalgearsloth) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:934]: Manually having to deadmin seems much more prone to bugs than just doing this in a central location and checking if they get a role attached?

Comment (LankLTE) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:934]: Not sure exactly how that would work, to be honest. If you have ideas I'm all for it, but given this is (essentially) how the cvar worked already it seems easiest to just do this.

**Comment by LankLTE** [comment]
Tests work on this now

**Comment by LankLTE** [comment]
Will be working on this soon.

**Pull request merged**